src/iidm/NodeBreakerVoltageLevel.cpp

```diff
@@ -325,6 +338,22 @@ void NodeBreakerVoltageLevel::reduceVariantArraySize(unsigned long number) {
     m_variants.reduceVariantArraySize(number);
 }
 
+void NodeBreakerVoltageLevel::removeInternalConnections(unsigned long node1, unsigned long node2) {
+    const auto& filter = [this, node1, node2](const unsigned long& edge) {
+        return !m_graph.getEdgeObject(edge) && m_graph.getVertex1(edge) == node1 && m_graph.getVertex2(edge) == node2;
```

Comment:
This is not equivalent to the java code. An internal connection is not an oriented edge, so you also have to check if
`m_graph.getVertex1(edge) == node2 && m_graph.getVertex2(edge) == node1`